### ISSUE TITLE
Non-obvious connection string issue with generating client / better help needed

### ISSUE THREAD

**Issue by lawik** [electric-sql/electric#732]
I tried running the generating client script stuff and got hit by:

```
larswikman@Sliver KnowLedge % npx electric-sql generate
Error: Command failed: npx prisma db pull --schema=".electric_migrations_tmp_7XQqz7/prisma/schema.prisma"
Error: P1000

Authentication failed against database server at `localhost`, the provided database credentials for `prisma` are not valid.

Please make sure to provide valid database credentials for the database server at `localhost`.


    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:526:35)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: 'npx prisma db pull --schema=".electric_migrations_tmp_7XQqz7/prisma/schema.prisma"'
}
generate command failed: Introspection script exited with error code: 1

```

The traceback made me think the thing was broken. Only after I clicked through to the the [Generator script](https://electric-sql.com/docs/api/generator) docs did I see the odd default values that don't line up with the electric sync service defaults. Are they matched to a Docker image or a quick start? I ran electric sync service from source.

**Comment by msfstef** [comment]
The generate command includes an option for specifying the address of both the sync service and the proxy (see ```--service``` and ```--proxy``` flags) , see: https://electric-sql.com/docs/api/cli#options

The sync service's port and proxy port and password must be configured via the env vars , see: https://electric-sql.com/docs/api/service#core-config

There's also a starter template you can use: https://electric-sql.com/docs/quickstart?setup=generator

**Comment by lawik** [comment]
Okay, sure. That is helpful. I did get through the problem.

The weird part was mostly that the generator script and the sync service do not use the same defaults for some reason.

**Comment by msfstef** [comment]
@lawik could you point me towards the differing defaults that were confusing?

One I see is the secure mode being enabled by default in the source but disabled when using the CLI but that was to reduce the friction to starting local-only development using Electric's TS client. That should be the main motivator behind any differences.

**Comment by lawik** [comment]
If you set up the client and sync backend, both from scratch. Do they connect successfully or not? That was the problem, the defaults were not the same between the two and it meant with a very basic setup it did not work until I figured out the mismatch. I no longer recall exactly what was diffing. But essentially I had to jump through additional hoops during a basic setup which I assume is not the intent.
